src/admin/resolvers/mutations/ScheduledItem.ts

```diff
@@ -3,23 +3,41 @@ import {
   createScheduledItem as dbCreateScheduledItem,
 } from '../../../database/mutations';
 import { ScheduledItem } from '../../../database/types';
-import { scheduledSurfaceAllowedValues } from '../../../shared/types';
+import {
+  ACCESS_DENIED_ERROR,
+  scheduledSurfaceAllowedValues,
+} from '../../../shared/types';
 import { ScheduledCorpusItemEventType } from '../../../events/types';
 import { UserInputError } from 'apollo-server';
 import { PrismaClientKnownRequestError } from '@prisma/client/runtime';
+import { AuthenticationError } from 'apollo-server-errors';
+import { IContext } from '../../context';
 
 /**
  * Deletes an item from the Scheduled Surface schedule.
  *
  * @param parent
  * @param data
- * @param db
+ * @param context
  */
 export async function deleteScheduledItem(
   parent,
   { data },
-  context
+  context: IContext
 ): Promise<ScheduledItem> {
+  // Need to fetch the item first to check access privileges.
+  // Could we make it easier/not hit the DB by passing the Scheduled Surface GUID
+  // as a required input to the delete mutation?
+  const item = await context.db.scheduledItem.findUnique({
+    where: { externalId: data.externalId },
+  });
+
+  // Check if the user can execute this mutation.
+  if (item && !context.authenticatedUser.canWrite(item.scheduledSurfaceGuid)) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
```

Comment:
should we check first if `item` exists or does the `findUnique()` throw an error if it doesn't find an entry?

The reason I ask is if we have this scenario where item is not found and user _**is**_ authenticated to write, the if condition would be: 

`if( false && !True) --> if(false)` it would skip this if block and  continue to delete it

Reply:
@Herraj , the `dbDeleteScheduledItem()` function called immediately after throwing the authentication error will actually throw a Not Found error if the scheduled item is nowhere to be found in the database. 

We could potentially move that check right into the resolver and place it earlier in the chain, before the auth check, but this becomes almost a philosophical question - do we want to let unauthorised users know if something is not in the database? Or do we want to throw the authorisation error first if their privileges are not sufficient? @jpetto, any thoughts?

Reply:
we should throw authorization errors before any others. it's an internal tool so the stakes are low, but i'd rather keep good security practices 😄